### ISSUE TITLE
False positive type error when associating Java object

### ISSUE THREAD

**Issue by borkdude** [borkdude/clj-kondo#677]
```
$ clj-kondo --lint - <<< '(assoc {} :key1 2 :key2 (java.util.Date.))'
<stdin>::: error: Insufficient input.
```
